### PR TITLE
Changes including Horovod integration, removed coordinator code and u…

### PR DESCRIPTION
**Pull request by karkadad**
…sed dataset APIs in place of feed dictionaries.

This pull request is based on the Mozilla Project DeepSpeech code, but has been integrated with Horovod and currently only supports training.
In order to aid scaling to a large number of CPU nodes, we've removed the coordinator code. We've also used dataset APIs in place of feed dictionaries.
This script has been extensively tested on CPUs. This script has not been tested on GPUs.
We've also implemented a dataset partitioning scheme based on the rank of the worker to minimize imbalance between workers processing audio of different source lengths during distributed training.
We've also added code for learning rate gradual warmup for large batch size training.
This pull request is targeted at developers interested in scaling DeepSpeech training to a large number of nodes using Horovod on CPUs. Please let us know what the best place to host these changes would be(including possibly in a separate branch). Thanks!

### CODE REVIEW

**Comment by kdavis-mozilla** [comment]
First of all, thanks!

Second, and this is just from a quick look, it appears as if your PR has not been rebased against the latest master. For example all of the progressbar code is not in you PR.

I'm wondering if you could rebase to get these recent changes in you PR before we review it for inclusion?

Thanks again!

**Comment by jageshmaharjan** [comment]
I was also working on horovod, but i couldn't finish the task as of other ongoing projects.  Excited to see horovod integration with Mozilla's Deepspeech :)

**Comment by karkadad** [comment]
Thanks for taking a look! We're testing the progressbar code with different MPI versions and should have it included in the PR soon.

**Comment by kdavis-mozilla** [comment]
@karkadad I'm going to close the PR for lack of activity. If you want to re-open the PR, feel free to.
